assets/app/features/upgrade/controllers/upgrade-nodes.controller.js

```diff
@@ -100,6 +158,33 @@
             vm.nodesUpgrade.totalNodes = response.data.upgraded_nodes + response.data.remaining_nodes;
             vm.nodesUpgrade.currentNodes = response.data.current_nodes;
             vm.nodesUpgrade.currentAction = response.data.current_node_action;
+            vm.nodesUpgrade.computeNodesPostponedFlag = response.data.compute_nodes_postponed;
+
+            // check for the case of postpone upgrade of compute nodes, after controller nodes were
+            // upgraded, step status would still be 'running' while substep status would be 'finished'
+            var subStep = response.data.current_substep;
+            var subStepStatus = response.data.current_substep_status;
+            var nodesSelected = response.data.nodes_selected_for_upgrade
+            if (subStep === 'controller_nodes' && subStepStatus === 'finished' && nodesSelected === 'controllers') {
+                vm.nodesUpgrade.running = false;
+                // show partially upgrade message
+                vm.nodesUpgrade.upgradeControllersDone = true;
+                upgradeStepsFactory.setUpgradeStep(2);
+                if (!vm.nodesUpgrade.computeNodesPostponedFlag && !vm.nodesUpgrade.resumeUpgrade) {
+                    upgradeFactory.setPostponeComputeNodes();
```

Comment:
Why is this done in `updateModel`?

Reply:
The obvious reason for that line being there was because it worked. For future references, if you think there's a better place for it, please don't hesitate to say so with some explanations. It would save both of us a lot of time responding back and forth between time zones. In the mean time, I will try to see where would be a better place for it.

Reply:
I asked like that because it was pretty surprising to me as `updateModel` is supposed to update the variables based on some server responses. I thought that triggering postpone action would be done based on some user action (click on "go to dashboard"?).

Reply:
When I discussed with @jsuchome about how/when to call the postpone/resume API, I was under the impression that the postpone API should be called right after the controller node upgrade was done. So I coded accordingly. I can see from your point of view that it could be done based on a user action. But can you rely on the user clicking the 'Go to Dashboard' button? What if the user just moves on to crowbar ui without clicking the button, because he thinks that he could just go to the crowbar ui himself and it would be the same end result? I'm not against your idea though. If you think that we should tie the postpone API to the 'Go to Dashboard' button, I can do that.

Reply:
@jsuchome please confirm what was the intended workflow for postpone. The CLI has explicit call to postpone and resume so I thought it should be triggered by some user action in the UI as well. However it will probably not be as explicit as in the CLI.

Reply:
I'm quite sure I wanted it to be done after explicit user action: When user clicks Go to Dashboard.
Otherwise it is not necessary - if controller nodes are upgraded, and user continues directly with upgrade of compute nodes, there's no need to postpone/resume.

It's actually written in the card: https://jira.prv.suse.net/browse/SCRD-3454 : _'Go to Dashboard' can be enabled only after successful upgrade of controller nodes. When user clicks it, API call of "nodes postpone" needs to be called so crowbar knows about the postponed state of compute nodes._

Reply:
@jsuchome I guess I didn't check that card as often as I should have. I will make the change as you specify.